core/trino-main/src/main/java/io/trino/Session.java

```diff
@@ -389,15 +388,38 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
                 protocolHeaders);
     }
 
-    public Session withDefaultProperties(Map<String, String> systemPropertyDefaults, Map<String, Map<String, String>> catalogPropertyDefaults)
+    public Session withDefaultProperties(Map<String, String> systemPropertyDefaults, Map<String, Map<String, String>> catalogPropertyDefaults, AccessControl accessControl)
     {
         requireNonNull(systemPropertyDefaults, "systemPropertyDefaults is null");
         requireNonNull(catalogPropertyDefaults, "catalogPropertyDefaults is null");
 
-        // to remove this check properties must be authenticated and validated as in beginTransactionId
-        checkState(
-                this.transactionId.isEmpty() && this.connectorProperties.isEmpty(),
-                "Session properties cannot be overridden once a transaction is active");
```

Comment:
In some places we assume session configuration doesn't change during transaction, e.g. in Hive connector.

Please add a test

- setup data with timesamp having nanosecond precision, using timestamp_precision=NANOSECONDS
- for X,Y precisions in (NANOSECONDS,MICROSECONDS,MILLISECONDS) x (NANOSECONDS,MICROSECONDS,MILLISECONDS) (`HiveTimestampPrecision.values()`)
  - in a transaction
    - set timestamp_precision=X
    - read data
    - set timestamp_precision=Y
    - read data

Reply:
@findepi I'm having difficulties to find a way to test the method I have changed.

As can be seen in the image below, the method `io.trino.Session#withDefaultProperties` is being called only from `io.trino.dispatcher.QueuedStatementResource#getStatus(io.trino.spi.QueryId, java.lang.String, long, io.airlift.units.Duration, javax.ws.rs.core.UriInfo, javax.ws.rs.container.AsyncResponse)`

![session_withDefaultProperties_call_hierarchy](https://user-images.githubusercontent.com/57260417/131572095-c9a5f9f4-8722-4958-ac93-8d04cdf71927.png)


Can you point me an existing test class in which I could add the test case that you've previously described?